projecttacoma/fqm-testify
# Generate Resource Buttons from Data Requirements

# Summary
Once a measure is uploaded and a test patient is created, a list of test resource buttons is now generated from the data requirements of the measure

## New Behavior
After uploading a measure and creating a test patient, you can now see a list of test resource buttons. The title of the button represents the resource type and the sub title is the name of each valueset included in the codeFilters on the data requirement

## Code Changes
 - Added new `TestResourcesDisplay` component
 - Added function `getDataRequirementFiltersString` to `util/fhir`
 - Updated `measureUploadState` to contain a `valueSetMap` field which maps valueset urls to their names for display purposes
 - Updated `useEffect` of MeasureUpload component to populate a valueSetMap object
 - Added unit testing

# Testing Guidance
 - Run `npm test`
 - Run `npm run dev`
 - Navigate to `http://localhost:3000`
 - Upload a measure
 - Create a test patient
 - Notice the test resources popup and ensure that all data requirements from the uploaded measure are represented
 - Upload a different measure and ensure the list updates
